tests/tasks/manage_test_interface.yml

```diff
@@ -27,7 +27,9 @@
   when: "type == 'veth' and state == 'present' and
          interface not in current_interfaces"
 - name: Set up veth as managed by NetworkManager
-  command: nmcli d set {{ interface }} managed true
+  command: |
+    nmcli c reload
+    nmcli d set {{ interface }} managed true
```

Comment:
Is this two separate commands?  If so, you'll have to use a `shell` instead of a `command`:
```yaml
shell: |
  set -euo pipefail
  nmcli c reload
  nmcli d set {{ interface }} managed true
```
Or, alternately, if `nmcli` can take a "compound" command - not sure about the syntax, something like
`nmcli "c reload" - "d set {{ interface }} managed true"` - then you could use a `command`

Reply:
Sorry, I missed it. I ran this test to see it was successful. But it seems it did not execute the task `Set up veth as managed by NetworkManager`. Removing my "approval".
$ tox -e qemu -- --image-file /path/to/CentOS-7-x86_64-GenericCloud.qcow2c --become --become-user root -- tests_bridge_nm.yml

Reply:
Yep, the task fails as follows, but it's ignored.
```
TASK [Set up veth as managed by NetworkManager] ********************************************************************************
task path: /path/to/linux-system-roles/network/tests/tasks/manage_test_interface.yml:29
fatal: [/path/to/CentOS-7-x86_64-GenericCloud.qcow2c]: FAILED! => {"changed": false, "cmd": "nmcli c reload nmcli d set testnic1 managed true", "msg": "[Errno 2] No such file or directory", "rc": 2}
...ignoring
```

Reply:
I updated the patch, thanks.

Reply:
Thank you for the discussion, @liangwen12year. I ran these two scripts in a row as you suggested.
tests_ethernet_initscripts.yml tests_ethernet_nm.yml

It worked just fine with rhel-guest-image-7.9-30.x86_64.qcow2, which has the NetworkManager package installed by default. But it turned out the package is not included in CentOS-7-x86_64-GenericCloud.qcow2c which I used for my testing first. That's why my test failed.

I noticed these tests explicitly install NetworkManager package.
```
tests/tests_regression_nm.yml
tests/tests_eth_pci_address_match_nm.yml
tests/tests_ethtool_coalesce_nm.yml
tests/tests_ethtool_features_nm.yml
tests/tests_ethtool_ring_nm.yml
tests/tests_provider_nm.yml
```
In the same manner, can we add a task to install `NetworkManager` to manage_test_interface.yml to make sure it is available before calling `nmcli`?

Reply:
I will, thanks for testing.

Reply:
Hi @nhosoi , I updated the patch again.